File: leetcode/src/test/java/org/luvx/leetcode/java/easy/_206/SolutionTest.java
package org.luvx.leetcode.java.easy._206;

import org.junit.jupiter.api.Test;
import org.luvx.leetcode.link.ListNode;
import org.luvx.leetcode.util.LinkUtils;

public class SolutionTest {

    @Test
    public void testReverseList2_1() {
        final Solution exec = new Solution();
        ListNode h = LinkUtils.initLinked(1, 2, 3, 4, 5);
        LinkUtils.printLink(
                h = exec.reverseList2(h)
        );
        LinkUtils.printLink(
                exec.reverseList2_1(h)
        );
    }
}

File: leetcode/src/test/kotlin/org/luvx/leetcode/java/easy/_637/SolutionTest.kt
package org.luvx.leetcode.java.easy._637;

import org.luvx.leetcode.TreeNode
import org.luvx.leetcode.out
import kotlin.test.BeforeTest
import kotlin.test.Test

class SolutionTest {
    private var root: TreeNode? = null

    @BeforeTest
    fun before() {
        root = TreeNode.of(
            5,
            TreeNode.of(
                4,
                TreeNode.of(
                    11,
                    TreeNode.of(7),
                    TreeNode.of(2)
                )
            ),
            TreeNode.of(
                8,
                TreeNode.of(13),
                TreeNode.of(
                    4, null,
                    TreeNode.of(1)
                )
            )
        )
    }

    @Test
    fun averageOfLevels() {
        val exec = Solution()
        exec.averageOfLevels(root).out()
    }
}